crates/polars-ops/src/chunked_array/list/namespace.rs

```diff
@@ -309,6 +309,11 @@ pub trait ListNameSpaceImpl: AsList {
                     let filtered_ca = ca.filter_with_func(&lambda_expressions)?;
                     Ok(filtered_ca.into_series())
                 },
+                DataType::Binary => {
+                    let ca = s_ref.binary()?;
```

Comment:
Does Series not provide a filter_with_func?
I think this maybe redundant code

Reply:
will check.

Reply:
implementation requires re-factoring the way we handle PolarsDataType and PolarsNumericType. I propose a new PR for this.